### PR TITLE
Add OIDC & API to Readme

### PR DESCRIPTION
**Pull request by FluidSense**
OIDC and API information could be nice for easier integration with OW4, so I added it to the readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,3 +120,59 @@ We highly recommend setting up your editor to automatically fix linting issues,
 ### Our recommendation
 
 In dotkom, we find that [PyCharm](https://www.jetbrains.com/pycharm/) is a great IDE that is well suited for contributing to onlineweb4. It'll come with support for ESLint, stylelint, and PEP 8 out of the box, and can be set up to run isort and Flake8 with some ease. We recommend it to our beginners who don't want to spend a lot of time setting up the plugins or extensions mentioned above, or don't have any preferences of their own yet. You can apply for a free student license [here](https://www.jetbrains.com/student/).
+
+## API  
+Onlineweb4 comes with an API located at `/api/v1`.  
+Autogenerated Swagger-docs can be found at `/api/v1/docs`.  
+Some endpoints require user authentication. See _Oauth/OIDC_.  
+
+## OAUTH/OIDC
```
Comment (henrikhorluck) [inline README.md:129]:
```suggestion
Some endpoints require user authentication. See [Oauth/OIDC](#OAuth/OIDC).  

## Oauth/OIDC  
```
https://gist.github.com/asabaylus/3071099

File: README.md [hunk 2]
```diff
@@ -120,3 +120,59 @@ We highly recommend setting up your editor to automatically fix linting issues,
...
 In dotkom, we find that [PyCharm](https://www.jetbrains.com/pycharm/) is a great IDE that is well suited for contributing to onlineweb4. It'll come with support for ESLint, stylelint, and PEP 8 out of the box, and can be set up to run isort and Flake8 with some ease. We recommend it to our beginners who don't want to spend a lot of time setting up the plugins or extensions mentioned above, or don't have any preferences of their own yet. You can apply for a free student license [here](https://www.jetbrains.com/student/).
+
+## API  
+Onlineweb4 comes with an API located at `/api/v1`.  
+Autogenerated Swagger-docs can be found at `/api/v1/docs`.  
+Some endpoints require user authentication. See _Oauth/OIDC_.  
+
+## OAUTH/OIDC  
+Onlineweb4 has an Oauth2/OIDC-provider built-in, allowing external projects to authenticate users through Onlineweb4.  
+[Oath: Which flow should I use?](https://auth0.com/docs/authorization/which-oauth-2-0-flow-should-i-use)
```
Comment (henrikhorluck) [inline README.md:131]:
```suggestion
[OAuth: Which flow should I use?](https://auth0.com/docs/authorization/which-oauth-2-0-flow-should-i-use)  
```

File: README.md [hunk 3]
```diff
@@ -120,3 +120,59 @@ We highly recommend setting up your editor to automatically fix linting issues,
...
+## API  
+Onlineweb4 comes with an API located at `/api/v1`.  
+Autogenerated Swagger-docs can be found at `/api/v1/docs`.  
+Some endpoints require user authentication. See _Oauth/OIDC_.  
+
+## OAUTH/OIDC  
+Onlineweb4 has an Oauth2/OIDC-provider built-in, allowing external projects to authenticate users through Onlineweb4.  
+[Oath: Which flow should I use?](https://auth0.com/docs/authorization/which-oauth-2-0-flow-should-i-use)  
+[Digital Ocean: Introduction to Oauth 2](https://www.digitalocean.com/community/tutorials/an-introduction-to-oauth-2)  
+[Oath: Authorization Flow with PKCE - Single Page Apps](https://auth0.com/docs/flows/authorization-code-flow-with-proof-key-for-code-exchange-pkce)
```
Comment (henrikhorluck) [inline README.md:133]:
```suggestion
[OAuth: Authorization Flow with PKCE - Single Page Apps](https://auth0.com/docs/flows/authorization-code-flow-with-proof-key-for-code-exchange-pkce)  
```

File: README.md [hunk 4]
```diff
@@ -120,3 +120,59 @@ We highly recommend setting up your editor to automatically fix linting issues,
...
+```  
+
+go to `localhost:8000/admin` and log in with your administrator user.  
+Scroll to `OpenID Connect Provider` and add a new `Client`.  
+Fill in the information related to the selected flow (see linked documentation on flows).
+Upon saving, a client-ID and client secret is generated for you. Revisit the client you created to retrieve the id and secret.  
+
+### Usage in a project  
+[Automated configuration and all endpoints](https://online.ntnu.no/openid/.well-known/openid-configuration)  
+There are many packages that provide oauth2/oidc-helpers. These can be quite helpful to automate much of the Oauth2-flow.
```
Comment (henrikhorluck) [inline README.md:153]:
```suggestion
There are many packages that provide oauth2/oidc-helpers. These can be quite helpful to automate much of the OAuth2-flow.
```

File: README.md (hunk 1 above)
Comment (henrikhorluck) [reply README.md:129]: Note: I am a bit unsure about case-sensitivity, but I checked it manually, and the `/` is removed it appears

File: README.md (hunk 2 above)
Comment (FluidSense) [reply README.md:131]: I meant to prefix the title with the authoring organization, Auth0 😅 
Not sure how that became Oath, but I will correct it

**Pull request merged**